src/Microsoft.Windows.CsWin32/templates/WinRTCustomMarshaler.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace Windows.Win32.CsWin32.InteropServices
+{
+    internal class WinRTCustomMarshaler : global::System.Runtime.InteropServices.ICustomMarshaler
+    {
+        private string winrtClassName;
+
+        private WinRTCustomMarshaler(string cookie)
+        {
+            this.winrtClassName = cookie;
+        }
+
+        /// <summary>
+        /// Gets an instance of the marshaler given a cookie
+        /// </summary>
+        /// <param name="cookie">Cookie used to create marshaler</param>
+        /// <returns>Marshaler</returns>
+        public static global::System.Runtime.InteropServices.ICustomMarshaler GetInstance(string cookie)
+        {
+            return new WinRTCustomMarshaler(cookie);
+        }
+
+        void global::System.Runtime.InteropServices.ICustomMarshaler.CleanUpManagedData(object ManagedObj)
+        {
+        }
+
+        void global::System.Runtime.InteropServices.ICustomMarshaler.CleanUpNativeData(global::System.IntPtr pNativeData)
+        {
+            global::System.Runtime.InteropServices.Marshal.Release(pNativeData);
+        }
+
+        int global::System.Runtime.InteropServices.ICustomMarshaler.GetNativeDataSize()
+        {
+            throw new global::System.NotImplementedException();
+        }
+
+        global::System.IntPtr global::System.Runtime.InteropServices.ICustomMarshaler.MarshalManagedToNative(object ManagedObj)
+        {
+            throw new global::System.NotImplementedException();
+        }
+
+        object global::System.Runtime.InteropServices.ICustomMarshaler.MarshalNativeToManaged(global::System.IntPtr pNativeData)
+        {
+            var assembly = typeof(global::Windows.Foundation.IMemoryBuffer).Assembly;
```

Comment:
What guarantees that `IMemoryBuffer` is generated? Or is this a WinRT type?

Reply:
It's a WinRT type.